app/event/raise-event.js

```diff
@@ -1,29 +1,13 @@
+import appInsights from 'applicationinsights'
 import { PublishEvent } from 'ffc-ahwr-event-publisher'
 import { eventQueue } from '../config/messaging.js'
 
-export const raiseEvent = async (event, status = 'success') => {
+export const raiseEvent = async (event, logger) => {
   try {
     const eventPublisher = new PublishEvent(eventQueue)
-
-    const eventMessage = {
-      name: event.name,
-      properties: {
-        id: event.id,
-        sbi: event.sbi,
-        cph: event.cph,
-        reference: event.reference,
```

Comment:
This was the actual cause of the bug, as the raise-ineligibility-event.js file wasn't putting reference at `event.reference` but putting it at `event.data.reference`.